src/join/composer.jl

```diff
@@ -640,36 +738,40 @@ function innerjoin(df1::AbstractDataFrame, df2::AbstractDataFrame;
                    makeunique::Bool=false,
                    validate::Union{Pair{Bool, Bool}, Tuple{Bool, Bool}}=(false, false),
                    renamecols::Pair=identity => identity,
-                   matchmissing::Symbol=:error)
+                   matchmissing::Symbol=:error,
+                   order::Union{Nothing, Symbol}=nothing)
     if !all(x -> x isa Union{Function, AbstractString, Symbol}, renamecols)
         throw(ArgumentError("renamecols keyword argument must be a `Pair` " *
                             "containing functions, strings, or `Symbol`s"))
     end
     return _join(df1, df2, on=on, kind=:inner, makeunique=makeunique,
                  indicator=nothing, validate=validate,
                  left_rename=first(renamecols), right_rename=last(renamecols),
-                 matchmissing=matchmissing)
+                 matchmissing=matchmissing, order=order)
 end
 
-innerjoin(df1::AbstractDataFrame, df2::AbstractDataFrame, dfs::AbstractDataFrame...;
-          on::Union{<:OnType, AbstractVector} = Symbol[],
-          makeunique::Bool=false,
-          validate::Union{Pair{Bool, Bool}, Tuple{Bool, Bool}}=(false, false),
-          matchmissing::Symbol=:error) =
-    innerjoin(innerjoin(df1, df2, on=on, makeunique=makeunique, validate=validate,
-                        matchmissing=matchmissing),
-              dfs..., on=on, makeunique=makeunique, validate=validate,
-              matchmissing=matchmissing)
+function innerjoin(df1::AbstractDataFrame, df2::AbstractDataFrame, dfs::AbstractDataFrame...;
+                   on::Union{<:OnType, AbstractVector} = Symbol[],
+                   makeunique::Bool=false,
+                   validate::Union{Pair{Bool, Bool}, Tuple{Bool, Bool}}=(false, false),
+                   matchmissing::Symbol=:error,
+                   order::Union{Nothing, Symbol}=nothing)
+    res = innerjoin(df1, df2, on=on, makeunique=makeunique, validate=validate,
+                        matchmissing=matchmissing)
+    for dfn in dfs
+        res = innerjoin(res, dfn, on=on, makeunique=makeunique, validate=validate,
+                        matchmissing=matchmissing, order=order)
```

Comment:
When there are multiple inputs and `order=:right`, we only need to pass it for the last call, right?

Reply:
right, fixed